Route manifest not compatible with Page.authenticate "redirectTo"

### What is the problem?

When I try to do
```
SomePage.authenticate = { redirectTo: Routes.LoginPage() }
```
I get the TypeScript error `error TS2322: Type 'RouteUrlObject' is not assignable to type 'string'.`

### What are detailed steps to reproduce this?

1. run `tsc`

### Run `blitz -v` and paste the output here:

```
Windows 10 | win32-x64 | Node: v14.15.5

blitz: 0.34.3 (global)
blitz: 0.35.0 (local) 

  Package manager: npm 
  System:
    OS: Windows 10 10.0.19042
    CPU: (6) x64 Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz
    Memory: 6.65 GB / 15.94 GB
  Binaries:
    Node: 14.15.5 - C:\Program Files\nodejs\node.EXE     
    Yarn: 1.22.10 - C:\Program Files\nodejs\yarn.CMD     
    npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  npmPackages:
    @prisma/client: ~2.20 => 2.20.1
    blitz: 0.35.0 => 0.35.0
    prisma: ~2.20 => 2.20.1
    react: 0.0.0-experimental-6a589ad71 => 0.0.0-experimental-6a589ad71
    react-dom: 0.0.0-experimental-6a589ad71 => 0.0.0-experimental-6a589ad71
    typescript: ~4.2 => 4.2.4
```

--- a/packages/core/src/blitz-app-root.tsx
+++ b/packages/core/src/blitz-app-root.tsx
@@ -68,12 +68,13 @@ export function withBlitzAppRoot(UserAppRoot: React.ComponentType<any>) {
         }
       } else {
         const authenticate = props.Component.authenticate
-        if (
-          authenticate &&
-          typeof authenticate === "object" &&
-          typeof authenticate.redirectTo === "string"
-        ) {
-          const url = new URL(authenticate.redirectTo, window.location.href)
+        if (authenticate && typeof authenticate === "object" && authenticate.redirectTo) {
+          let {redirectTo} = authenticate
+          if (typeof redirectTo !== "string") {
+            redirectTo = formatWithValidation(redirectTo)
+          }
+
+          const url = new URL(redirectTo, window.location.href)
           url.searchParams.append("next", window.location.pathname)
           window.location.replace(url.toString())
         }

--- a/packages/core/src/types.ts
+++ b/packages/core/src/types.ts
@@ -38,7 +38,7 @@ export interface AppProps<P = {}> extends NextAppProps<P> {
 }
 export type BlitzPage<P = {}, IP = P> = NextPage<P, IP> & {
   getLayout?: (component: JSX.Element) => JSX.Element
-  authenticate?: boolean | {redirectTo?: string}
+  authenticate?: boolean | {redirectTo?: string | RouteUrlObject}
   suppressFirstRenderFlicker?: boolean
   redirectAuthenticatedTo?: string | RouteUrlObject
 }

